src/test/java/edu/oakland/test/middleware01/MiddlewareGisManagerStub.java

```diff
@@ -0,0 +1,24 @@
+package edu.oakland.test.middleware01;
+
+import edu.oakland.helper.admin.Satellite;
+import edu.oakland.production.middleware01.MiddlewareGisManager;
+
+public class MiddlewareGisManagerStub implements MiddlewareGisManager {
+    
+  public String receiveGpsSignalStrength(Satellite satelliteSignal) {
+    return satelliteSignal.getSatelliteName();
+  }
+    
+  public void storeLocationDataPoint() {
+    
+  }
+    
+  public String attemptToRegainGpsLock(String satelliteName) {
+    return "";
+  }
+    
+  public String evaluateGpsSignalStrength(Satellite satellite) {
+    return "GPS0";
```

Comment:
This shouldn't be hard coded, but should instead be used to verify that basically the correct satellite was sent in. (Basically the same as how you do it above.)
```suggestion
    return satellite.getSatelliteName();
```